src/common/getAgesOfFamily.ts

```diff
@@ -0,0 +1,50 @@
+import { youngestAdultBirthYear } from "@/app/clients/form/birthYearDropdown";
+import { getCurrentMonth, getCurrentYear } from "@/common/date";
+
+export const isChildUsingBirthYear = (birthYear: number): boolean => {
+    return getCurrentYear() - birthYear <= 16;
+};
+
+export const getChildAgeUsingBirthYearAndMonth = (
+    birthYear: number,
+    birthMonth: number | null,
+    shortHand: boolean
+): string => {
+    const years = getCurrentYear() - birthYear;
+    let totalMonths;
+    if (birthMonth) {
+        const months = getCurrentMonth() - birthMonth;
+        totalMonths = years * 12 + months;
+    } else {
+        totalMonths = years * 12;
+    }
+    if (totalMonths <= 1) {
+        if (shortHand) {
+            return `${totalMonths}m`;
+        }
+        return `${totalMonths}-month-old`;
+    }
+    if (totalMonths <= 12) {
+        if (shortHand) {
+            return `${totalMonths}m`;
+        }
+        return `${totalMonths}-months-old`;
+    }
+    if (shortHand) {
+        return `${years}`;
+    }
+    return totalMonths < 24 ? `${years}-year-old` : `${years}-years-old`;
```

Comment:
```suggestion
    const years = getCurrentYear() - birthYear;
    const months = birthMonth ? getCurrentMonth() - birthMonth : 0;
    const totalMonths = years * 12 + months;
    
    if (totalMonths <= 1) {
        return shortHand ? `${totalMonths}m` : `${totalMonths}-month-old`;
    }
    if (totalMonths <= 12) {
        return shortHand ? `${totalMonths}m` : return `${totalMonths}-months-old`;
    }
    if (shortHand) {
        return `${years}`;
    }
    return totalMonths < 24 ? `${years}-year-old` : `${years}-years-old`;
```

Reply:
I'd double check that suggestion works as intented but hopefully the shorter version is a bit easier to read

Some other bits
- should it be `totalMonths < 12`? I'd put it as 1-year-old
- should the shorthand for above 2 years be `${years}y` (with a y at the end)?

And generally how much does the age matter for adults? Because without the month the estimates might be inaccurate, e.g. If I put in 2000 as the year but my birthday is in September it will say I'm 24 but that's incorrect. But I don't know how much that matters at all

Reply:
Thanks Kristy - It does look a lot easier to read! 

The only scenarios that may be relevant that I can think of is if they need it for safeguarding children (boundary between 15/16 years old) and or young adults. I don't actually know if this is the case but I do know that agencies that interact with children usually have these. Originally we did think about creating a boundary for children eg. 15-16 years old but it may have looked very crowded on the client details modal if they have a lot of children.

Reply:
I guess you can't really decide without exact birth date - you don't want to mark a 15 year old as an adult but you don't want to mark an independent 16 year old as a child I think
Does this actually cause problems anywhere? I imagine if there's no coded in part which changes much for this boundary the workers/volunteers can be sensible about it

Reply:
Only in the client and parcel modals and shopping list where it shows how many adults/children there are with their respective ages.